tests/test_diffraction_objects.py

```diff
@@ -369,6 +371,23 @@ def test_all_array_setter():
         actual_do.all_arrays = np.empty((4, 4))
 
 
+def test_id_getter():
+    do = DiffractionObject()
+    assert hasattr(do, "id")
+    assert isinstance(do.id, UUID)
+    assert len(str(do.id)) == 36
```

Comment:
for testing `id`, check the the id length,  its type, its existence.

Reply:
Here I would probably mock it

Reply:
Yes, I've maintained this test since we still want to have a getter test for `id` but also added a mock right below.